src/main/java/org/apache/commons/crypto/cipher/Openssl.java

```diff
@@ -323,11 +622,42 @@ public void clean() {
         }
     }
 
+    /**
+     * Continues a multi-part update of the Additional Authentication
+     * Data (AAD).
+     * <p>
+     * Calls to this method provide AAD to the cipher when operating in
+     * modes such as AEAD (GCM).  If this cipher is operating in
+     * either GCM mode, all AAD must be supplied before beginning
+     * operations on the ciphertext (via the {@code update} and
+     * {@code doFinal} methods).
+     *
+     * @param aad the buffer containing the Additional Authentication Data
+     *
+     */
+    public void updateAAD(byte[] aad) {
```

Comment:
Is it right that updateAAD simply overwrite the aadbuffer?

I thought it means append more aad data.